src/App.tsx

```diff
@@ -17,18 +34,29 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                handleSelect={handleSelect}
+                handleQuery={handleQuery}
+                clearQuery={clearQuery}
+                query={query}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {isLoading && <Loader />}
+              {isError && <div>Error</div>}
+              {todos && !query && <TodoList todos={todos[selector as keyof typeof todos]} />}
+              {todos
+                && query
+                && (
+                  <TodoList
+                    todos={todos[selector as keyof typeof todos].filter((todo) => todo.title.toLowerCase().includes(query.toLowerCase()))}
```

Comment:
move logic from pros to a separate variable or function
Also, move `query.toLowerCase()` from filter,  don't do this on every filter iteration